mwdb/resources/oauth.py

```diff
@@ -0,0 +1,238 @@
+        identity = OpenIDUserIdentity.query.filter(
+            OpenIDUserIdentity.sub_id == userinfo["sub"]
+        ).one()
+        if identity is None:
+            raise Forbidden("Unknown identity")
+
+        user = identity.user
+
+        if user.pending:
+            raise Forbidden(
+                "User registration is pending - "
+                "check your e-mail inbox for confirmation"
+            )
+
+        if user.disabled:
+            raise Forbidden("User account is disabled.")
+
+        user.logged_on = datetime.datetime.now()
+        db.session.commit()
+
+        auth_token = user.generate_session_token()
+
+        logger.info("User logged in", extra={"login": user.login})
+        schema = AuthSuccessResponseSchema()
+        return schema.dump(
+            {
+                "login": user.login,
+                "token": auth_token,
+                "capabilities": user.capabilities,
+                "groups": user.group_names,
+            }
+        )
+
+
+class OpenIDBindAccountResource(Resource):
+    @requires_authorization
+    def post(self, provider_name):
+        """
+        ---
+        summary: Bind existing mwdb account with OpenID identity
+        description: |
+            TODO
+        tags:
+            - auth
+        parameters:
+            - in: path
+              name: provider_name
+              schema:
+                type: string
+              description: OpenID provider name.
+        requestBody:
+            description: OpenID Connect request state
+            content:
+              application/json:
+                schema: OpenIDAuthorizeRequestSchema
+        responses:
+            200:
+                description: When OpenID identity was successively bound to mwdb account
+        """
+        provider = (
+            db.session.query(OpenIDProvider)
+            .filter(OpenIDProvider.name == provider_name)
+            .first()
+        )
+        if not provider:
+            raise NotFound(f"Requested provider name '{provider_name}' not found")
+
+        schema = OpenIDAuthorizeRequestSchema()
+        obj = loads_schema(request.get_data(as_text=True), schema)
+        redirect_uri = f"{app_config.mwdb.base_url}/oauth/authorize"
+        userinfo = provider.fetch_id_token(
+            obj["code"], obj["state"], obj["nonce"], redirect_uri
+        )
+        identity = OpenIDUserIdentity(
+            sub_id=userinfo["sub"], provider_id=provider.id, user_id=g.auth_user.id
+        )
+        db.session.add(identity)
+        db.session.commit()
+
+        logger.info(
+            "Oauth identity was successively authenticated",
+            extra={"user": g.auth_user.login, "provider": provider.name},
+        )
+
+
+class OpenIDAccountIdentitiesResource(Resource):
+    @requires_authorization
+    def get(self):
+        """
+        ---
+        summary: List OpenID bound external identities
+        description: |
+            TODO
+        tags:
```

Comment:
`security: bearerAuth` is required, so Swagger knows that Authorization header should be included
```suggestion
        security:
            - bearerAuth: []
        tags:
```